### ISSUE TITLE
'files' subcommand should allow wildcards

### ISSUE THREAD

**Issue by oniony** [oniony/TMSU#19]
In an email, Victoria Stuart asked:

> Does tmsu support wildcard expressions?  E.g., for tagging? searches?

To which I had to reply 'no'. From what I gathered Victoria wanted to be able to do:

    $ tmsu files she*

To list files that match any of the tags that match the pattern, e.g. 'sheep', 'shenanigans', 'shellac'. Or, when tagging a file:

    $ tmsu tag somefile car*

To tag a file 'carpet', 'cardigan' and 'carpel-tunnel-syndrome'.

This suggests she is trying to build some sort of structure with the tags so perhaps tag hierarchies need to be considered or thought about. Personally I've always felt that it would be better to use tag intersections would be a better way of doing this. For example, instead of defining the following tags:

    * car-hatchback-mercedes
    * car-saloon-audi
    * van-volkswagen

It would be better to define:

    * car
    * van
    * hatchback
    * saloon
    * make=mercedes
    * make=audi
    * make=volkswagen

And to apply the tags as appropriate.

**Comment by student-sx** [comment]
I think this would be a nice feature if you want to use tmsu within emacs helm: https://emacs.stackexchange.com/a/34836/2323 

If I call such a helm source it should first display all files typing substrings of tags should narrow the list of candiates down dynamically. For example

>pattern: foo bar

Should display all results including foo and bar, i.e. files tagged like Afoo and Xbar or fooX and barB. 

If I then refine it to 

>pattern: fooX bar

It just matches files with tags (fooX and barB) or (fooX and Xbar) if you take the example from above. 

I think a helm source working like this would be possible if tmsu would provide wildcards.

**Comment by glindsey** [comment]
Sorry to necro a five-year-old (!) issue, but there's another place wildcards would be extremely useful: searching for titles. 

Example: Say you have a video file containing "The Eternal Sunshine of the Spotless Mind". So you tag it `title="The Eternal Sunshine of the Spotless Mind"`. Much later, you want to find it; you don't remember the full name, though, so you want to search for `title="*spotless*"`. Now, you could say that you should just rename the file to contain the title, and in this case that might be a viable solution, but it would still be helpful for other string metadata tags.

Anyways, just a thought. I absolutely adore this project, and am really grateful for the work you've put into it!

**Comment by oniony** [comment]
Agree, this would be useful. I guess it would make sense to have a separate operator for this. I've seen tilde (~) used in other systems to do fuzzy 'contains' matches, such as Jira, so I think this would work:

    tmsu files title ~ spotless
    tmsu files title ~ "spotless mind"
